### PR TITLE
add start chat, disable hint to terminal instance context menu

### LINKED ISSUE #213813: The terminal initial hint should be easily toggled by right clicking on the hint
Can this support right click to toggle the setting to hidden?

image

### PR DESCRIPTION
**Pull request by meganrogge**
fixes #213813 
fixes #213817

To do: 
- only include disable hint action in menu when widget is visible
- change context key to track that instead of when it's enabled
- ⚡ maybe just right click on the hint element to disable it

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalMenus.ts [hunk 1]
```diff
@@ -184,6 +184,29 @@ export function setupTerminalMenus(): void {
...
+			{
+				id: MenuId.TerminalInstanceContext,
+				item: {
+					command: {
+						id: TerminalCommandId.DisableInitialHint,
+						title: localize2('disableInitialHint', 'Disable Initial Hint'),
+					},
+					group: '1_copilot@0',
+					order: 2,
+					// when: ContextKeyExpr.and(TerminalContextKeys.initialHintEnabled),
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalMenus.ts:207]: this doesn't do anything when uncommented. very strange as debugging and `dev: inspect context keys` shows it's updated correctly. 

even when I set the initial value of the context key to `true`, it's not showing up if I include a `when`. Maybe the context key is registered after this menu is created?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalMenus.ts:207]: nope, debugging shows the context key is created before the menu

**Review by Tyriar — CHANGES_REQUESTED** [review]
We want a sub-menu like in the editor:

Screenshot 2024-06-04 at 11 02 01 AM

**Comment by meganrogge** [comment]
Have a PR to add that, but unclear how / if we can add a core contributed command, to disable the hint, to that submenu. For now, just putting it in the instance menu.

**Comment by meganrogge** [comment]
that's for the best anyway because this initial hint might be used for more than just chat

**Pull request closed without merging**